test/AWS.Deploy.CLI.IntegrationTests/ServerMode/DependencyValidationOptionSettings.cs

```diff
@@ -0,0 +1,207 @@
+                    {
+                        {"ApplicationIAMRole.CreateNew", "false"}
+                    }
+                });
+
+                applicationIAMRole = (await restClient.GetConfigSettingsAsync(sessionId)).OptionSettings.First(x => x.Id.Equals("ApplicationIAMRole"));
+                Assert.Equal(ValidationStatus.Valid, applicationIAMRole.Validation.ValidationStatus);
+                Assert.True(string.IsNullOrEmpty(applicationIAMRole.Validation.ValidationMessage));
+                validCreateNew = applicationIAMRole.ChildOptionSettings.First(x => x.Id.Equals("CreateNew"));
+                Assert.Equal(ValidationStatus.Valid, validCreateNew.Validation.ValidationStatus);
+                Assert.True(string.IsNullOrEmpty(validCreateNew.Validation.ValidationMessage));
+                validRoleArn = applicationIAMRole.ChildOptionSettings.First(x => x.Id.Equals("RoleArn"));
+                Assert.Equal(ValidationStatus.Invalid, validRoleArn.Validation.ValidationStatus);
+                Assert.Equal("Invalid IAM Role ARN. The ARN should contain the arn:[PARTITION]:iam namespace, followed by the account ID, and then the resource path. For example - arn:aws:iam::123456789012:role/S3Access is a valid IAM Role ARN. For more information visit https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_identifiers.html#identifiers-arns",
+                    validRoleArn.Validation.ValidationMessage);
+
+                applyConfigResponse = await restClient.ApplyConfigSettingsAsync(sessionId, new ApplyConfigSettingsInput()
+                {
+                    UpdatedSettings = new Dictionary<string, string>()
+                    {
+                        {"ApplicationIAMRole.CreateNew", "true"}
+                    }
+                });
+
+                applicationIAMRole = (await restClient.GetConfigSettingsAsync(sessionId)).OptionSettings.First(x => x.Id.Equals("ApplicationIAMRole"));
+                Assert.Equal(ValidationStatus.Valid, applicationIAMRole.Validation.ValidationStatus);
+                Assert.True(string.IsNullOrEmpty(applicationIAMRole.Validation.ValidationMessage));
+                validCreateNew = applicationIAMRole.ChildOptionSettings.First(x => x.Id.Equals("CreateNew"));
+                Assert.Equal(ValidationStatus.Valid, validCreateNew.Validation.ValidationStatus);
+                Assert.True(string.IsNullOrEmpty(validCreateNew.Validation.ValidationMessage));
+                validRoleArn = applicationIAMRole.ChildOptionSettings.First(x => x.Id.Equals("RoleArn"));
+                Assert.Equal(ValidationStatus.Valid, validRoleArn.Validation.ValidationStatus);
+                Assert.True(string.IsNullOrEmpty(validRoleArn.Validation.ValidationMessage));
+            }
+            finally
+            {
+                cancelSource.Cancel();
+                _stackName = null;
+            }
+        }
+
+        private async Task WaitTillServerModeReady(RestAPIClient restApiClient)
```

Comment:
This is already duplicated several times, could we move to something shared?

More generally, I think it'd be nice if server mode tests had less boilerplate so it'd be easier to see what the test is actually doing. Though not necessarily introduced by this PR, so okay if we move to a new issue.

Reply:
Done